lib/mobile/mobile-core/appetize-device-discovery.ts

```diff
@@ -0,0 +1,64 @@
+import { EventEmitter } from "events";
+import { DEVICE_DISCOVERY_EVENTS } from "../../constants";
+import { deviceEmitter } from "cloud-device-emulator";
+import { AppetizeDevice } from "../device/appetize-device";
+import { values } from "lodash";
+
+export class AppetizeDeviceDiscovery extends EventEmitter implements Mobile.IDeviceDiscovery {
+	private devices: IDictionary<Mobile.IDevice> = {};
+	private _hasStartedLookingForDevices = false;
+
+	constructor(private $injector: IInjector) {
+		super();
+	}
+
+	public addDevice(device: Mobile.IDevice) {
+		this.devices[device.deviceInfo.identifier] = device;
+		this.raiseOnDeviceFound(device);
+	}
+
+	public removeDevice(deviceIdentifier: string) {
+		let device = this.devices[deviceIdentifier];
+		if (!device) {
+			return;
+		}
+
+		delete this.devices[deviceIdentifier];
+		this.raiseOnDeviceLost(device);
+	}
+
+	public async startLookingForDevices(): Promise<void> {
+		if (!this._hasStartedLookingForDevices) {
+			this._hasStartedLookingForDevices = true;
+			values(deviceEmitter.getCurrentlyAttachedDevices()).forEach(basicInfo => {
+				this.addAppetizeDevice(basicInfo);
+			});
+
+			deviceEmitter.on(DEVICE_DISCOVERY_EVENTS.DEVICE_FOUND, (basicInfo: IAppetizeDeviceBasicInfo) => {
+				this.addAppetizeDevice(basicInfo);
+			});
+
+			deviceEmitter.on(DEVICE_DISCOVERY_EVENTS.DEVICE_LOST, (basicInfo: IAppetizeDeviceBasicInfo) => {
+				const device: Mobile.IDevice = this.$injector.resolve(AppetizeDevice, { basicInfo: basicInfo });
```

Comment:
why do you need this, can't you just pass `basicInfo.identifier` to `removeDevice`